api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -340,6 +340,42 @@ public static void validateWriteSchema(Schema tableSchema, Schema writeSchema,
     }
   }
 
+  /**
+   * Validates whether the provided schema is compatible with the expected schema.
+   *
+   * @param ctx the validation context used in the error message
+   * @param expectedSchema the expected schema
+   * @param actualSchema the actual schema
+   * @param checkNullability whether to check nullability
+   * @param checkOrdering whether to check field ordering
+   */
+  public static void validateSchema(String ctx, Schema expectedSchema, Schema actualSchema,
+                                    Boolean checkNullability, Boolean checkOrdering) {
+    List<String> errors;
+    if (checkNullability) {
+      errors = CheckCompatibility.writeCompatibilityErrors(expectedSchema, actualSchema, checkOrdering);
+    } else {
+      errors = CheckCompatibility.typeCompatibilityErrors(expectedSchema, actualSchema, checkOrdering);
+    }
+
+    if (!errors.isEmpty()) {
+      StringBuilder sb = new StringBuilder();
+      String header = String.format("Schema is not compatible with expected %s schema:", ctx);
+      sb.append(header)
+          .append("\nexpected schema:")
```

Comment:
I don't think this is needed since the last line ends with `expected %s schema:`

Reply:
Fixed.